logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt

```diff
@@ -194,23 +238,23 @@ class SyncManagerImpl(
 
     override suspend fun waitUntilSlowSyncCompletion() {
         startSyncIfIdle()
-        syncRepository.syncState.first { it is SyncState.ProcessingPendingEvents || it is SyncState.Live }
+        syncRepository.syncState.first { it in setOf(SyncState.GatheringPendingEvents, SyncState.Live) }
```

Comment:
Just out of curiosity, we are guaranteed that` syncRepository.syncState.first` will always have at least one element ? I'm not sure if by any chance this could throw something like `NoSuchElementException` 🤔

Reply:
Yes. It's a `StateFlow` (requires an initial value) and is initialized with a default value of `WAITING`.